src/Microsoft.Health.Fhir.Core/Features/Operations/ICosmosDBAccessTokenProvider.cs

```diff
@@ -0,0 +1,23 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Health.Fhir.Core.Features.Operations
+{
+    public interface ICosmosDBAccessTokenProvider
```

Comment:
Should this also provide the TokenCredential to the FhirCosmosClientInitializer?

Reply:
Very good point, updated the interface and FhirCosmosClientInitializer